### PR TITLE
Add `microsoft_` prefix to all packages

### LINKED ISSUE #48: naming of the packages: add a Microsoft prefix to the package names
have a "microsoft" prefix? i.e. microsoft_kiota_abstractions instead of kiota_abstractions

This enables:
- consitency: java com.microsoft.kiota, dotnet microsoft.kiota, go github.com/microsoft/kiota, JS @microsoft/kiota, ....
- security: at some point for npm and maven, the security team worked with the feed owners and said "any package identified as Microsoft MUST have the microsoft prefix in the name, and we'll lock down who can publish to that prefix"... and we had to go through painful migration process.

### PR DESCRIPTION
**Pull request by ricardoboss**
Fixes #48.

This adds a `microsoft_` prefix to all packages, all relative dependencies and links to indicate these packages are owned/managed by Microsoft and possibly prevent us from having to migrate to any given prefix in the future.

### CODE REVIEW

**Comment by joanne-ter-maat** [comment]
The config files in the .run folder still refer to the old names/paths.

**Comment by ricardoboss** [comment]
> The config files in the .run folder still refer to the old names/paths. 

Right. I wanted to remove them anyways. You can just delete the whole folder.

**Review by joanne-ter-maat — APPROVED** [review]

File: packages/microsoft_kiota_http/example/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+name: microsoft_kiota_http_example
```
Comment (baywet) [inline packages/microsoft_kiota_http/example/pubspec.yaml:1]: should we move the example outside of the http package structure?

**Review by baywet — APPROVED** [review]

File: packages/microsoft_kiota_http/example/pubspec.yaml (hunk 1 above)
Comment (ricardoboss) [reply packages/microsoft_kiota_http/example/pubspec.yaml:1]: No that is part of the Dart package layout. Each package _should_ come with an example on how to use it. In theory, we should also add examples to all other packages.

Comment (baywet) [reply packages/microsoft_kiota_http/example/pubspec.yaml:1]: Thanks for explaining this to me. :)

**Pull request merged**